### PR TITLE
Update rust toolchain to nightly 1.66.0-nightly

### PR DESCRIPTION
**Pull request by rahulku**
### Description of changes: 

Update rust tool chain to new nightly build. One expected result was modified for a new line number, and other files were updated for escaping '<' for rustdoc.

* How is this change tested? kani-regressions

* Is this a refactor change? no

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs [hunk 1]
```diff
@@ -506,10 +506,10 @@ impl<'tcx> GotocCtx<'tcx> {
         self.codegen_ty(typ).to_pointer()
     }
 
-    /// A reference to a Struct<dyn T> { .., data: T} is translated to
+    /// A reference to a Struct\<dyn T\> { .., data: T} is translated to
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:509]: Would
 ```
`String<dyn T> {.., data: T}`
```
work here?

**Review by danielsn — APPROVED** [review]

File: kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs [hunk 2]
```diff
@@ -90,7 +90,7 @@ impl<'tcx> GotocCtx<'tcx> {
         RAW_PTR_FROM_BOX.iter().fold(box_expr, |expr, name| expr.member(name, &self.symbol_table))
     }
 
-    /// Box<T> initializer
+    /// Box\<T\> initializer
```
Comment (celinval) [inline kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs:93]: Same.

File: kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs (hunk 1 above)
Comment (tedinski) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:509]: I think I'd prefer these to be updated to use code blocks instead, e.g.:

```
/// A reference to a `Struct<dyn T> { .., data: T}` is translated to
```

But either way works for me

Comment (rahulku) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs:509]: good call. changing.

**Review by celinval — APPROVED** [review]
Just a minor comment. How do you get to do such an easy update? @adpaco-aws will agree with me on that one. :)

File: rust-toolchain.toml [hunk 3]
```diff
@@ -2,5 +2,5 @@
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 
 [toolchain]
-channel = "nightly-2022-10-11"
-components = ["llvm-tools-preview", "rustc-dev", "rust-src", "rustfmt"]
+channel = "nightly-2022-10-25"
+components = ["llvm-tools-preview", "rustc-dev", "rust-src", "rustfmt"]
```
Comment (tedinski) [inline rust-toolchain.toml:6]: nit: your editor might not be configured to emit trailing newlines

for vscode it looks like I added `"files.insertFinalNewline": true,` to its "user settings"

File: kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs (hunk 1 above)
Comment (rahulku) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs]: done

Comment (rahulku) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/typ.rs]: done

File: kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs (hunk 2 above)
Comment (rahulku) [reply kani-compiler/src/codegen_cprover_gotoc/utils/utils.rs]: done

File: rust-toolchain.toml (hunk 3 above)
Comment (rahulku) [reply rust-toolchain.toml:6]: thanks - i did change that.

**Pull request merged**